src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -89,9 +136,7 @@ public bool MoveNext(CancellationToken cancellationToken = default(CancellationT
                 }
                 catch (Exception ex) when (RetryabilityHelper.IsResumableChangeStreamException(ex))
                 {
-                    var newCursor = _changeStreamOperation.Resume(_binding, cancellationToken);
-                    _cursor.Dispose();
-                    _cursor = newCursor;
+                    _cursor = ResumeCursor(_cursor, cancellationToken);
```

Comment:
Do this instead:

    var newCursor = Resume(cancellationToken);
    _cursor.Dispose();
    _cursor = newCursor;

It is better for the call to Dispose to be in the same method where we replace the old value of _cursor with a new one.

Reply:
Done